src/components/ProjectListItem.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { Box, Grid, Paper, Typography, useMediaQuery, useTheme } from '@mui/material';
+import { Project } from '../types';
+import OasisApprovedIcon from '../assets/OasisApprovedIcon.svg';
+import TagsList from './TagsList';
+
+interface ProjectListItemProps {
+  project: Project;
+  handleProjectClick: (project: Project) => void;
+  selectedTags: string[];
+  handleTagClick: (tag: string) => void;
+  allTags: string[];
+}
+
+const truncateText = (text: string, maxLength: number) => {
+  if (text.length > maxLength) {
+    return text.slice(0, maxLength) + '…';
+  }
+  return text;
+};
+
+const ProjectListItem: React.FC<ProjectListItemProps> = ({
+  project,
+  handleProjectClick,
+  selectedTags,
+}) => {
+  const theme = useTheme();
+  const isMobileScreen = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const combinedTags = [...project.tags, ...project.languages];
+  return (
+    <Grid item xs={12} sm={6} md={4} key={project.name}>
```

Comment:
`key={project.name}` doesn't help here because parent ProjectListItem doesn't have a unique key.
So add a unique key to ProjectListItem:
```tsx
        {filteredAndSortedProjects.map((project) => (
              <ProjectListItem
                key={project.name}
```
and then this one can be removed `<Grid item xs={12} sm={6} md={4}>`